Marlin/src/gcode/geometry/G92.cpp

```diff
@@ -93,10 +93,37 @@ void GcodeSuite::G92() {
       COPY(coordinate_system[active_coordinate_system], position_shift);
   #endif
 
-  bool didA = parser.seenval('A');
-  if (didA) {
-    const float curr_z = current_position[Z_AXIS];
-    const float adjust_z = parser.value_axis_units(Z_AXIS);
+  bool doZAdjust = false;
+  float adjust_z = 0.0;
+
+
+
+  if(parser.seenval('A')){
+    adjust_z = parser.value_axis_units(Z_AXIS);
+    doZAdjust = true;
+  }else if(parser.seenval('B')){
```

Comment:
For "G92 B", adjustment is calculated as if it was "G92 A" and flag is set. Then the same block of code handles both ZTO moves.